src/files/files_client.py

```diff
@@ -0,0 +1,479 @@
+"""
+DeepOrigin FilesClient Module
+
+This module provides a client for interacting with the DeepOrigin file service.
+It offers high-level methods for file operations such as upload, download,
+listing directories, and synchronizing files between local and remote storage.
+
+Example usage:
+    ```python
+    from deeporigin.files import FilesClient
+    
+    # Create a client (uses default authentication if token not provided)
+    client = FilesClient()
+
+    # List files in a directory
+    files = client.list_dir('files:///data/', flag="recursive")
+    print(f"Found {len(files)} files")
```

Comment:
if we use the client pattern (that we use in the platform API and datahub API) we can simplify this even more. instead of methods that require the instantiation of a class, we simply have a function that looks like:

```py

def list_dir(..., client=FilesClient()):
    pass
```

this way, users don't need to know about FilesClient, and this allows us to pass a MockClient object that makes testing easier.

Reply:
it's inefficient/seems poor design to construct a client every time. in the future we may want to move connection validation checks, API version checks etc into constructor

Reply:
> it's inefficient/seems poor design to construct a client every time.

exactly. that's what i'm suggesting. in the example above, you can call 

```py
list_dir(...)
```

and not pass a client